### DISCUSSION TITLE
Example test case of HW8

### DISCUSSION THREAD

**Discussion by 33577** [prosyslab-classroom/cs524-program-analysis#442] [Q&A] [answered]
### Your Name

Jonguk Jeon

### Question

안녕하세요, 기본 예시로 나오는 example1.c가 왜 ill-typed인지 궁금합니다. 
어떤 부분이 문제인지 설명해주실 수 있나요? 
감사합니다. 

``` example1.c
int f(int x) { return x + 1; }
int main() {
  int x = 10;
  int y = 10 + x;
  int z = x - 10;
  int w = y / z;
  char v = 10 + 2;
  if (v == 10)
    w++;
  else
    w--;
  y = f(x);
  return 0;
}
```

(eng) 

Hi, I wonder why example1.c is ill-typed. Is there anyone can tell me why? 
Thanks

**Comment by Re-st** [comment, marked as answer]
안녕하세요.
이전에 비슷한 질문이 있어 공유드립니다.
https://github.com/prosyslab-classroom/cs524-program-analysis/discussions/391 가 답변이 될까요?

**Reply by 33577** [reply]
감사합니다!

**Reply by 33577** [reply]
확인하고 왔는데, %div를 대신 저장하는 것이 왜 타입 문제를 일으키는지 모르겠습니다. 
%div나 %v나 모두 int여서 타입적으로 문제 없는 것 아닌가요? 

` storing 12 at %v, %div is stored at %v, which causes the type error.`

**Reply by goodtaeeun** [reply]
예제에서 `%v`는 `i8` 타입인데, 여기에 `i32` 타입인 `%div`를 저장하는 것이 타입 에러입니다.

**Reply by goodtaeeun** [reply]
아래는 `example1.ll`의 일부입니다. 도움이 될 것 같아 첨부합니다.

```llvm
%v = alloca i8, align 1
%div = sdiv i32 %2, %3
store i8 %div, i8* %v, align 1
```

**Reply by 33577** [reply]
아 이해했습니다! 감사합니다!!! (_ _ )
